### PR TITLE
Fix n+1 and integration test

### PR DESCRIPTION
**Pull request by SaboorSohaib**
# Hi Dear :wave: 

**In this project, I implemented the following.**

- [ ] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- [ ] Use Capybara to write integration tests for each view in your project.

  - User index page.
  - User show page.
  - User post index page.
  - Post-show page.

### Screenshots:

 **:diamonds: Before:**
load before
load before1

**:diamonds: After:**
load-after
load-after1

## Thank You

### CODE REVIEW

File: spec/features/index.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      visit users_path
+      expect(page).to have_content(@user1.posts_counter)
+      expect(page).to have_content(@user2.posts_counter)
+    end
+
+    it 'has correct link' do
+      visit users_path
+      expect(page).to have_link(@user1.name, href: user_path(@user1))
+      expect(page).to have_link(@user2.name, href: user_path(@user2))
+    end
```
Comment (TedLivist) [inline spec/features/index.html.erb_spec.rb:34]: - [ ] You have done well adding this test however this is not how we should check for the redirect scenario. We need to actually **`click`** the button/link in this test and ensure it redirects to the user show page as noted in the requirement.

File: spec/features/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    it 'show user bio' do
+      expect(page).to have_content(@user1.bio)
+    end
+
+    it 'dsiplay 3 most recent post' do
+      expect(page).to have_content(@post3.title)
+      expect(page).to have_content(@post2.title)
+      expect(page).to have_content(@post1.title)
+    end
```
Comment (TedLivist) [inline spec/features/show.html.erb_spec.rb:43]: - [ ] For the user show page you are missing the following test scenarios:

  - Tests that asserts you can see the number of posts the user has written.
  - Tests that asserts you can see a button that lets me view all of a user's posts.
  - Tests that redirect to the post's show page when you click on a user's post.
  - Tests that redirect to the user's post's index page when you click on a see all posts.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @SaboorSohaib and @billymohajeri,

Good job so far!
✔️ Passing and well-written tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @SaboorSohaib, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- PR is professional :+1: 
- All written unit tests are passing :heavy_check_mark: 
- All the requested tests scenarios have been covered :+1: 

## Optional suggestions

- [ ] The MIT license link is broken. Let's kindly fix it in order to render the readme file professional

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SaboorSohaib** [comment]
### Hi @FrederickMih 
Thank you very much for reviewing and approving my project. :+1:

**Pull request merged**